src/python/WMCore/Database/MongoDB.py

```diff
@@ -36,27 +37,48 @@ def __init__(self, database=None, server=None, port=None, replicaset=None,
                       if missing (TODO)
         :logger:      Logger
         """
-        self.server = server # '127.0.0.1'
-        self.port = port # 8230
+        self.server = server
+        self.port = port
         self.logger = logger
         self.mockMongoDB = mockMongoDB
         if mockMongoDB and mongomock is None:
             msg = "You are trying to mock MongoDB, but you do not have mongomock in the python path."
             self.logger.critical(msg)
             raise ImportError(msg)
+
+        # NOTE: We need to explicitely check for server availiability.
+        #       From pymongo Documentation:
+        #       """
+        #           ...
+        #           Starting with version 3.0 the :class:`MongoClient`
+        #           constructor no longer blocks while connecting to the server or
+        #           servers, and it no longer raises
+        #           :class:`~pymongo.errors.ConnectionFailure` if they are
+        #           unavailable, nor :class:`~pymongo.errors.ConfigurationError`
+        #           if the user's credentials are wrong. Instead, the constructor
+        #           returns immediately and launches the connection process on
+        #           background threads.
+        #           ...
+        #       """
         try:
             if mockMongoDB:
                 self.client = mongomock.MongoClient()
                 self.logger.info("NOTICE: MongoDB is set to use mongomock, instead of real database.")
             elif replicaset:
-                self.client = MongoClient(self.server, self.port, replicaset=replicaset, **kwargs )
+                self.client = MongoClient(host=self.server, port=self.port, replicaset=replicaset, **kwargs )
             else:
-                self.client = MongoClient(self.server, self.port, **kwargs)
+                self.client = MongoClient(host=self.server, port=self.port, **kwargs)
             self.client.server_info()
+            self.client.admin.command('ping')
+        except ConnectionFailure as ex:
+            msg = "Could not connect to MongoDB server: %s. Server not available. \n"
+            msg += "Giving up Now."
+            self.logger.error(msg, self.server)
+            raise ex
```

Comment:
To avoid chained exceptions, maybe implement it as `raise ex from None`

Reply:
Good idea. It would be helpful also in the comment few lines above. I am adding this now.